src/lib/components/DeckGLMap/layers/wells/wellsLayer.ts

```diff
@@ -0,0 +1,55 @@
+import { CompositeLayer } from "@deck.gl/core";
+import { CompositeLayerProps } from "@deck.gl/core/lib/composite-layer";
+import { GeoJsonLayer, GeoJsonLayerProps } from "@deck.gl/layers";
+import { RGBAColor } from "@deck.gl/core/utils/color";
+
+export interface WellsLayerProps<D> extends CompositeLayerProps<D> {
+    pointRadiusScale: number;
+    lineWidthScale: number;
+    outline: boolean;
+}
+
+interface WellDataType {
+    type: string;
+    geometry: Record<string, unknown>;
+    properties: {
+        name: string;
+        color: RGBAColor;
+    };
+}
+
+function _getFillColor(d): RGBAColor {
+    return d.properties.color;
+}
+
+export default class WellsLayer extends CompositeLayer<
+    unknown,
+    WellsLayerProps<unknown>
+> {
+    renderLayers(): GeoJsonLayer<unknown>[] {
+        const properties: GeoJsonLayerProps<unknown> = {
+            id: "outline",
+            data: this.props.data,
+            pickable: true,
+            pointRadiusUnits: "pixels",
+            lineWidthUnits: "pixels",
+            pointRadiusScale: this.props.pointRadiusScale,
+            lineWidthScale: this.props.lineWidthScale,
+        };
+
+        const outline = new GeoJsonLayer(this.getSubLayerProps(properties));
+
+        properties.id = "colors";
+        properties.getFillColor = _getFillColor;
```

Comment:
Since it's a tiny function it could be a lambda instead of a free function:
`properties.getFillColor = (d: WellDataType): RGBAColor => d.properties.color;`